Backend/src/main/java/xmlteam4/Project/controllers/CoverLetterController.java

```diff
@@ -1,22 +1,41 @@
 package xmlteam4.Project.controllers;
 
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
 import org.springframework.http.MediaType;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 import sun.reflect.generics.reflectiveObjects.NotImplementedException;
+import xmlteam4.Project.services.CoverLetterService;
 
 @RestController
 @RequestMapping("/cover-letters")
 public class CoverLetterController {
 
+    @Autowired
+    private CoverLetterService coverLetterService;
+
     @GetMapping(value = "/{id}", produces = MediaType.TEXT_HTML_VALUE)
     public ResponseEntity<String> getCoverLetter(@PathVariable("id") String id) {
-        throw new NotImplementedException();
+        String cover = "";
+        try {
+            cover = coverLetterService.findOne(id);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        return new ResponseEntity<>(cover, HttpStatus.OK);
```

Comment:
Method should be:

`
try {
    return new ResponseEntity<>(coverLetterService.findOne(id), HttpStatus.OK);
} catch (Exception e) {
    return new ResponseEntity<>("Failed to find cover letter with given id", HttpStatus.NOT_FOUND)
}
`
Same goes for others. Use appropriate status codes!

Reply:
Methods changed accordingly